src/gui/folderman.cpp

```diff
@@ -1631,4 +1682,43 @@ void FolderMan::restartApplication()
     }
 }
 
+void FolderMan::slotSetupPushNotifications(const Folder::Map &folderMap)
+{
+    for (auto folder : folderMap) {
+        const auto account = folder->accountState()->account();
+
+        // See if the account already provides the PushNotifications object and if yes connect to it.
+        // If we can't connect at this point, the signals will be connected in slotPushNotificationsReady()
+        // after the PushNotification object emitted the ready signal
+        slotConnectToPushNotifications(account.data());
+        connect(account.data(), &Account::pushNotificationsReady, this, &FolderMan::slotConnectToPushNotifications, Qt::UniqueConnection);
+    }
+}
+
+void FolderMan::slotProcessFilesPushNotification(Account *account)
+{
+    qCInfo(lcFolderMan) << "Got files push notification run etag job if possible";
+
+    for (auto folder : _folderMap) {
+        // Just run on the folders that belong to this account
+        if (folder->accountState()->account() != account) {
+            continue;
+        }
+
+        qCInfo(lcFolderMan) << "Run etag job if possible on " << folder;
+        runEtagJobIfPossible(folder);
```

Comment:
So indeed we could have an issue there, if the folder is syncing already or blocked by another one we might loose the event. This is both good and bad. Good because it avoids syncing too many times for nothing if you receive a burst of such notifications, bad because you might miss a change (I'd say chances are slim but will increase with the amount of files in the folder).

I think we might want to call `scheduleFolder` or `scheduleFolderNext` here instead.

Reply:
I would go for `scheduleFolder()` because it does not add the folder to the queue if it's already added.

Reply:
Sounds fine.